aries_vcx/src/protocols/issuance/holder/state_machine.rs

```diff
@@ -517,6 +515,7 @@ pub async fn create_credential_request(
 
 async fn _make_credential_request(
     wallet_handle: WalletHandle,
+    thread_id: String,
```

Comment:
It seems that `&str` would be sufficient and avoid cloning, same for `my_pw_did` argument

Reply:
I can make this &str, but it doesn't really help, because this is used to set 
```
CredentialRequest::create()
        .set_thread_id(thread_id)
```
and `set_thread_id` run clone anyway. I'd suggest in another PR we change macros like `threadlike_optional` etc. to require `String` instead of `&str`, as it's anyway ends up cloning the `str`. I think this will then trickle-up on over layers of calling code, as well and will probably lead to some unification/clearing of how we pass `&str`/`String`, as it's fairly inconsistent now.